### ISSUE TITLE
High Sierra Deploy.sh Support

### ISSUE THREAD

**Issue by ZombieTheBest** [syscl/XPS9350-macOS#157]
Hi all, 
I have edited Deploy.sh to fix some High Sierra errors and hoping to provide a more stable patching for the new OS. I have already sent a Pull request, meanwhile if you want to try out the new script (remember this is still BETA and could broke your installation), you can find it here: https://github.com/ZombieTheBest/XPS9350-10.13-HighSierra

**Note:** current Syscl script creates a Clipboard issue where clipboard doesn't work anymore on QHD models. This is caused by the ClockPixel Patch. This Deploy use another patch for that (CoreDisplayFixUp). If you already have the clipboard issue then you will need to restore the vanilla CodeDisplay.framework (I will attach mine here), or do a Clean Install. 

Also, if you need basic NTFS R/W support: https://github.com/ZombieTheBest/macOS-ntfs

Let me know if it's anything wrong with the script, thank you!
[CoreDisplay.framework.zip](https://github.com/syscl/XPS9350-macOS/files/1413963/CoreDisplay.framework.zip)

**Comment by chenyisen** [comment]
Thanks for your working. I've tried the script and it prompted some wrong code:
./Deploy.sh: line 154: contains command not found
./Deploy.sh: line 823 syntax error near unexpected token `}'
./Deploy.sh: line 823: `}'

I'm just a prime user, and I don't have much knowledge on programming. But profited by your and syscl's contribution, my laptop could use macOS. Thanks again

**Comment by ZombieTheBest** [comment]
@chenyisen Thank you for your feedback. Try the new one please let me know if it fixes your issue 👍

**Comment by chenyisen** [comment]
Oops.. I still can't paste anything. Maybe I need to get a new install of High Sierra? :(

**Comment by ZombieTheBest** [comment]
Yes, that's caused by the previous Pixel Clock Patch. Only way to fix it is
Clean install or restore the vanilla CoreDisplay.framework (attached in the
first post).

No more errors btw? :)

Il 25 Ott 2017 12:00 PM, "chenyisen" <notifications@github.com> ha scritto:

**Comment by chenyisen** [comment]
I've got a clean install of 10.13, but there are some other wrong codes:
./Deploy.sh: line 837: syntax error near unexpected token `}'

./Deploy.sh: line 837: `}'
 I guess the dawn of victory is in front of me!!

**Comment by ZombieTheBest** [comment]
@chenyisen Please try new deploy and report back! 👍

**Comment by ZombieTheBest** [comment]
@chenyisen Please try last Deploy and let me know! Thank you 👍

**Comment by chenyisen** [comment]
The script run correctly. But clipboard is also offline......
But I noticed that during "Unlocking maximum pixel clock...", it prompted that
./Deploy.sh: line 2077: [: missing `]

Maybe this code caused my failure?

**Comment by ZombieTheBest** [comment]
@chenyisen Sh** a space was missing. Fixed, sorry for the inconvenience. Unfortunately the missing space made the script exec the wrong PixelCode-Section. Try the Deploy again first, and see if you see any other error, then please restore vanilla CoreDisplay.framework provided in post 1 to fix clipboard issue!

**Comment by chenyisen** [comment]
./Deploy.sh line 837:syntax error near unexpected token `}'

./Deploy.sh: line 837: `}'
It also prompted this wrong code.......

btw, I replaced your CoreDisplay.framework to System/Library/Framework, and rebuild permission, but my clipboard is offline stiffly...

**Comment by ZombieTheBest** [comment]
@chenyisen Try now please, it should be all fixed. Regarding the CoreDisplay.framework, run installer again (without clean install) and it should replace it with the vanilla one. 

Please try again the script and report back any issue before proceeding.
Thank you very much for your time

**Comment by chenyisen** [comment]
Congratulations! My clipboard finally worked! However, it need a clean install of 10.13. Thanks for your hardworking!

**Comment by ZombieTheBest** [comment]
@chenyisen That's a good news! 👍 
If you have any other issue with High Sierra please let me know and will update the script

**Comment by chenyisen** [comment]
Everything seems working well. But my handoff not seems working. I can find iPhone on wifi-selection menu, but I can't connect on it. And my apps opened on iPhone couldn't appear on my dock. Did you encountered this problem？

**Comment by ZombieTheBest** [comment]
I had to disable Handoff patch because it was causing problem on my
hackintosh (random reboots). In my case Handoff is still working. You can
try to enable handoff patch in your clover file (under kernel and kexts
patches) using clover configurator.

Il 26 Ott 2017 15:34, "chenyisen" <notifications@github.com> ha scritto:

Everything seems working well. But my handoff not seems working. I can find
iPhone on wifi-selection menu, but I can't connect on it. And my apps
opened on iPhone couldn't appear on my dock. Did you encountered this
problem？

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/syscl/XPS9350-macOS/issues/157#issuecomment-339667518>,
or mute the thread

**Comment by chenyisen** [comment]
I didn’t choose disable of the handoff patches, and I replaced HEX to my broad ID. And now I can find hotspots of my iPhone, but I can’t connect via Instant Hotspot. Do you have any resolvant?

发自我的 iPhone

在 2017年10月26日，下午9:37，Mirco Ianese <notifications@github.com<mailto:notifications@github.com>> 写道：

I had to disable Handoff patch because it was causing problem on my
hackintosh (random reboots). In my case Handoff is still working. You can
try to enable handoff patch in your clover file (under kernel and kexts
patches) using clover configurator.

Il 26 Ott 2017 15:34, "chenyisen" <notifications@github.com<mailto:notifications@github.com>> ha scritto:

Everything seems working well. But my handoff not seems working. I can find
iPhone on wifi-selection menu, but I can't connect on it. And my apps
opened on iPhone couldn't appear on my dock. Did you encountered this
problem？

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/syscl/XPS9350-macOS/issues/157#issuecomment-339667518>,
or mute the thread

**Comment by ZombieTheBest** [comment]
You don't have to edit anything in handoff patch on config.plist

Il 27 Ott 2017 04:04, "chenyisen" <notifications@github.com> ha scritto:

**Comment by chenyisen** [comment]
[CLOVER.zip](https://github.com/syscl/XPS9350-macOS/files/1421947/CLOVER.zip)
Here is my clover folder. I'm glad if it could provide any help.
I tested my Handoff and Instant Hotspot, I found that I can make a call via iPhone when connect same wifi. But the functions bluetooth needed is failed, such as universal clipboard, and icons opened on iPhone doesn't appear on my dock.

**Comment by ZombieTheBest** [comment]
@chenyisen Thank you. Please remove SSDT-USBX and SSDT-PNLF from CLOVER/ACPI/Patched as they shouldn't be needed. 
Regarding Handoff, patch should be correct. Please disable any other WiFi Patch in config.plist and put inside the last WiFi patch you will find in repo config.plist (that is for 10.13).
